pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionRuntimeManager.java

```diff
@@ -537,6 +454,223 @@ private void stopFunction(String fullyQualifiedInstanceId, boolean restart) thro
         }
     }
 
+    /**
```

Comment:
For all new functionality, could we collapse getInstanceStats and getFunctionStats into one implementation? I see a lot of duplicate code

Reply:
Can't really combine those two function since getFunctionInstanceStats returns the stats for instance with getFunctionStats returns stats overall and for all instances.  Different objects are returned.  What can be refactored / combined is the logic between the get stats functions and get status functions.  I didn't want to refactor the logic there since there are already many changes in this PR as it is.

Reply:
sounds good